From d82457ce5ca7455e336da5e244d95f90e52aaef8 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Tue, 18 Apr 2017 01:17:26 -0700
Subject: [PATCH] Makefile: libacl should depend on include

Fixed race issue:
 In file included from acl_copy_entry.c:22:0:
 libacl.h:19:21: fatal error: sys/acl.h: No such file or directory
  #include <sys/acl.h>

[snip]

 compilation terminated.
 acl_get_file.c:27:24: fatal error: acl/libacl.h: No such file or directory
  #include <acl/libacl.h>
                         ^

The acl.h is in "include" directory, and include/Makefile creates
symlink "sys" and "acl" poinst to current dirctory:
$ ls include/ -l
acl -> .
sys -> .

So if "libacl" target runs before "include", the error would happen
since no "acl" or "sys" directory.

Let libacl depend on include can fix the problem.

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index dce32d3..8a79379 100644
--- a/Makefile
+++ b/Makefile
@@ -48,7 +48,7 @@ else
 endif
 
 # tool/lib dependencies
-libacl: libmisc
+libacl: include libmisc
 getfacl setfacl chacl: libacl
 
 ifeq ($(HAVE_BUILDDEFS), yes)
-- 
2.10.2

